### ISSUE TITLE
cpuid can not be installed under user root

### ISSUE THREAD

**Issue by ffallah** [ocaml/opam#3438]
When I try to install cpuid under user root I get the following error:
```
#opam install cpuid
[WARNING] Running as root is not recommended
The following actions will be performed:
  - install cpuid 0.1.1

<><> Gathering sources ><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
[cpuid.0.1.1] found in cache

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
[ERROR] The compilation of cpuid failed at "/root/.opam/opam-init/hooks/sandbox.sh build ocaml
        pkg/pkg.ml build --pinned false --tests false".

#=== ERROR while compiling cpuid.0.1.1 ========================================#
# context     2.0.0~rc3 | linux/x86_64 | ocaml-base-compiler.4.06.1 | https://opam.ocaml.org/2.0#9f513703
# path        ~/.opam/4.06.1/.opam-switch/build/cpuid.0.1.1
# command     ~/.opam/opam-init/hooks/sandbox.sh build ocaml pkg/pkg.ml build --pinned false --tests false
# exit-code   1
# env-file    ~/.opam/log/cpuid-382-05296d.env
# output-file ~/.opam/log/cpuid-382-05296d.out
### output ###
# ocamlfind ocamlopt -package unix -package ocamlbuild -linkpkg -package ocb-stubblr myocamlbuild.ml /root/.opam/4.06.1/lib/ocamlbuild/ocamlbuild.cmx -o myocamlbuild
# [WARNING] Running as root is not recommended
# [ERROR] /root/.opam exists, but does not appear to be a valid opam root. Please remove it and use `opam init', or specify a different `--root' argument
# Failure: Ocb_stubblr: error running opam.
# pkg.ml: [ERROR] cmd ['ocamlbuild' '-use-ocamlfind' '-classic-display' '-j' '4' '-tag' 'debug'
#      '-build-dir' '_build' '-plugin-tag' 'package(ocb-stubblr)' 'opam'
#      'pkg/META' 'CHANGES.md' 'LICENSE.md' 'README.md' 'src/cpuid.a'
#      'src/cpuid.cmxs' 'src/cpuid.cmxa' 'src/cpuid.cma' 'src/cpuid.cmx'
#      'src/cpuid.cmi' 'src/cpuid.mli' 'src/dllcpuid_stubs.so'
#      'src/libcpuid_stubs.a']: exited with 2



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+- The following actions failed
| - build cpuid 0.1.1
+-
- No changes have been performed
```
But it works when I try it under another username.
I believe it is related to this issue : https://github.com/ocaml/opam/issues/3434 (Sandbox: opam root not visible)

**Comment by dra27** [comment]
This is a duplicate of #3434, as you note. You can "fix" this locally by adding `/root` to line 26 of `/root/.opam/opam-init/hooks/sandbox.sh` to give:

```sh
add_mounts ro /usr /bin /lib /lib32 /lib64 /etc /opt /nix/store /home /root
```

However, as @rjbou notes in #3434, the true fix needs to take place in ocb-stubblr which should determine the opam root using `OPAM_SWITCH_PREFIX` not `opam config var prefix`. I've submitted a PR for it which you can use by running `opam pin add ocb-stubblr git+https://github.com/dra27/ocb-stubblr.git#sandboxing`

**Comment by rjbou** [comment]
Resolved via #3540. Thanks!
